src/payment/strategies/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -0,0 +1,192 @@
+        const { paymentIntent, error } = await stripeJs.confirmPayment({
+            elements,
+            redirect: 'if_required',
+        });
+
+        if (error) {
+            if (error.type === 'card_error' || error.type === 'invalid_request_error' || error.type === 'validation_error') {
+                throw new Error(error.message);
+            }
+            throw new RequestError();
+        }
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+
+        const paymentPayload = {
+            methodId,
+            paymentData: {
+                formattedPayload: {
+                    credit_card_token: { token: paymentIntent ? paymentIntent.id : '' },
+                    vault_payment_instrument: shouldSaveInstrument,
+                    confirm: false,
+                },
+                shouldSetAsDefaultInstrument,
+            },
+        };
+
+        return await this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._unmountElement();
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    private _getInitializeOptions(options?: StripeUPEPaymentInitializeOptions): StripeUPEPaymentInitializeOptions {
+        if (this._initializeOptions) {
+            return this._initializeOptions;
+        }
+        if (options) {
+            this._initializeOptions = options;
+
+            return this._initializeOptions;
+        }
+
+        throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+    }
+
+    private async _getStripeElements(elementOptions?: StripeElementsOptions): Promise<StripeElements> {
+        if (this._stripeElements) {
+            return this._stripeElements;
+        }
+        if (elementOptions) {
+            const stripeJS = await this._getStripeJs();
+            this._stripeElements = stripeJS.elements(elementOptions);
+
+            return this._stripeElements;
+        }
+
+        throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+    }
+
+    private async _getStripeJs(stripePublishableKey?: string, stripeConnectedAccount?: string): Promise<StripeUPEClient> {
+        if (this._stripeUPEClient) {
+            return this._stripeUPEClient;
+        }
+
+        if (stripePublishableKey && stripeConnectedAccount) {
+            this._stripeUPEClient = await this._stripeScriptLoader.load(
+                stripePublishableKey,
+                stripeConnectedAccount,
+                this._locale
+            );
+
+            return this._stripeUPEClient;
+        }
+
+        throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+    }
+
+    private async _mountCardFields(clientToken: string = ''): Promise<StripeElement> {
```

Comment:
I was doing this as a workaround of Typescript because the Type definition of PaymentMethod has the clientToken defined as optional, but for stripeUPE we should always be getting a clientToken.